### DISCUSSION TITLE
"Cannot read properties of undefined" when deploying Prisma app to Vercel

### DISCUSSION THREAD

**Discussion by nktruong** [prisma/prisma#13661] [Q&A] [answered]
I have successfully built my Minecraft app in Sveltekit, but it only works locally. I run 'svelte-kit dev' and the site shows up:

![Screenshot of Shroomhearth's shopping directory](https://user-images.githubusercontent.com/88857875/172236327-42e8d231-d4ee-4ace-8d80-f1118d856cca.png)

I have it currently connected to Vercel. When it gets deployed, I get an error of:

```
500
Cannot read properties of undefined (reading 'shop')
TypeError: Cannot read properties of undefined (reading 'shop')
```

Screenshot on Vercel deployed site:

![500 status error code on website](https://user-images.githubusercontent.com/88857875/172236818-b0bc0b04-e43a-45df-af79-f2c578763995.png)

I'm pretty new to Prisma and SQLite coming from MongoDB. Let me know if any extra information or files are needed to diagnose the issue.

**Comment by janpio** [comment]
Is `shop` one of your models?
Did you run `prisma generate` explicitly during the deploy somewhere?
Can you deploy and also overwrite the cache on Vercel?

**Reply by nktruong** [reply]
Hi Jan, thanks for the response! Yes, `shop` is one of my models.

I have not run `prisma generate` during the deploy. I have added it now to my `build` script, so that it now runs `svelte-kit build && pnpx prisma generate`. It is now generating the Prisma Client, but still shows the same error on the website.

Here is the build log: https://gist.github.com/nktruong/9544da43168166070abfd5a7c26cbfa6 and this is the repo: https://github.com/nktruong/shroomshops/

I'll try overwriting the cache. According to the Vercel Docs for [Static Files Caching](https://vercel.com/docs/concepts/edge-network/caching), it says that the cache is invalidated when the project is redeployed with the last version. I tried deleting it and then reconnecting it to my repo, but no change.

**Comment by grantpitt** [comment, marked as answer]
I had a similar issue while using prisma, sveltekit, and vercel. The only thing that fixed it for me was adding a `postbuild` script in my `package.json` :
```json
"postbuild": "cp node_modules/@prisma/engines/*query* .vercel_build_output/functions/node/render/;cp prisma/schema.prisma .vercel_build_output/functions/node/render/"
```
I'm not totally sure what purpose it serves or why it's needed.

**Reply by pantharshit00** [reply]
Hello @nktruong! Sorry for the late follow up here. Did you get a chance to checkout this workaround?

Also, @grantpitt do you still need to do this workaround in the latest version?

**Reply by nktruong** [reply]
Hello @pantharshit00! It's been a long time since I've worked on this project. I'll keep this workaround in mind, thanks @grantpitt!

**Reply by grantpitt** [reply]
@pantharshit00 No, the latest version works great without this workaround.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
